### ISSUE TITLE
aws-vpc-cni chart document running multiple instances

### ISSUE THREAD

**Issue by laupow** [aws/amazon-vpc-cni-k8s#2000]
**What would you like to be added**:
Documentation about deploying multiple instances of the `aws-vpc-cni` chart. 

**Why is this needed**:
We run 4 different instances of the CNI application in our cluster. The only differences between the `DaemonSet` instances are labels, node selectors / affinities, and the env config.

We want to stop maintaining a bunch of custom manifests and move to helm charts. 

A bit of documentation on best practices for running >1 instance of the helm chart would be helpful. (ie `CustomResourceDefinition` deployment considerations)

**Comment by jayanthvn** [comment]
@laupow - Can you please expand more on the use case? Is my understanding correct, you have 4 different flavors of aws-node running on a cluster with different configs? To maintain the configs you would like a guide on setting up multiple instances of helm chart?

**Comment by laupow** [comment]
Yes, of course. 

We have a cluster with 7 different node groups. 

4 node groups share the same `aws-vpc-cni` default configuration.
2 node groups have a custom `aws-vpc-cni` configuration that only varies by env var.
1 node group has a custom `aws-vpc-cni` configuration and a collection of `ENIConfig` objects for per-AZ sub
nets/security groups.

I'd love to use the EKS managed add-ons, but last time I checked I couldn't configure it or run a different version on a node group. Instead we use nodeSelectors & tolerations to ensure the correct `aws-vpc-cni` daemonset gets scheduled to a given nodegroup. 

_I think_ it makes sense to deploy the helm chart as:
* one default `aws-vpc-cni` helm chart that owns the CRD
* 3 other `aws-vpc-cni` charts with env vars overrides configured as needed, but without the CRD, and with ENIConfigs

I asked about guidance because the helm chart seems maintained and a good fit for our needs. ~But it isn't mentioned in AWS docs as the preferred way to do a custom install of of `aws-vpc-cni`.~ Nevermind! I just looked to find the link and now it [defaults](https://docs.aws.amazon.com/eks/latest/userguide/managing-vpc-cni.html#updating-vpc-cni-add-on) to help charts 🙏  

The docs used to recommend this command as the official way:

```
kubectl apply -f https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.11.0/config/master/aws-k8s-cni.yaml
```

but that wasn't really practical since we need to customize those manifests.

Anyway, it looks like the chart is well-built and documented now. So we just need to transition over. Thanks.
